packages/docusaurus-theme-classic/src/theme/DocPage/styles.module.css

```diff
@@ -78,9 +78,3 @@
     ) !important;
   }
 }
-
-@media (max-width: 996px) {
-  .docSidebarContainer {
```

Comment:
In fact,  collapse sidebar button is already hidden, so I'm even more surprised how it can be displayed when the viewport is resized in browser.

https://github.com/facebook/docusaurus/blob/master/packages/docusaurus-theme-classic/src/theme/DocSidebar/styles.module.css#L97-L98